### PR TITLE
Developer Guide: overall architecture diagram (very very high level)

### LINKED ISSUE #96: Developer Guide: Overall Architecture Diagram

### PR DESCRIPTION
**Pull request by Leeyp**
overall architecture diagram (very very high level)

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -2,8 +2,26 @@
...
+Given below is a quick overview of each component.
+
+**[`FridgeFriend`](https://github.com/AY2021S2-CS2113-T10-1/tp/blob/master/src/main/java/seedu/fridgefriend/FridgeFriend.java)** is driven by a method called `Main`. 
+It is responsible for,
+* At app launch: Initializes the components in the correct sequence, and connects them up with each other.
+* At shut down: Shuts down the components and invokes cleanup methods where necessary.
+
+[**`Food`**](#common-classes) represents a collection of classes used by multiple other components.
+
+The rest of the App consists of four components.
```
Comment (kwokyto) [inline docs/DeveloperGuide.md:24]: Can we define our app as such:
- Food
- Command
- Exception
- Utilities
- FridgeFriend --> entry point

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -4,25 +4,43 @@
...
-The rest of the App consists of four components.
 
-* [**`UI`**](#ui-component): The UI of the App.
-* [**`Parser`**](#parser-component): Processes user input, and determines the related command to be executed.
-* [**`Command`**](#command-component): The command executor.
-* [**`Storage`**](#storage-component): Reads data from, and writes data to, the hard disk.
+The rest of the App consists of four components.
 
+* [**`Utilities`**](#utilities-component): The main component containing the key driver classes in 
+  FridgeFriend, such as `Parser`, `UI`, and `Storage`.
```
Comment (kwokyto) [inline docs/DeveloperGuide.md]: I think you can just briefly mention that it contains key driver classes. the other details will be in the utilities section

File: docs/diagrams/ArchitectureDiagram.puml [hunk 3]
```diff
@@ -5,26 +5,27 @@
...
+Utilities -[#blue]-> Exception
+Food -[#grey]> Utilities
+Command -> Food
+Command -> FridgeFriend
 
-Storage .right[STORAGE_COLOR].>File
-User ..> UI
+
+Utilities .right[STORAGE_COLOR].>File
+User ..> FridgeFriend
```
Comment (kwokyto) [inline docs/diagrams/ArchitectureDiagram.puml]: actually the user interacts with utilities not fridgefriend

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (Leeyp) [reply docs/DeveloperGuide.md]: Resolved

File: docs/diagrams/ArchitectureDiagram.puml (hunk 3 above)
Comment (Leeyp) [reply docs/diagrams/ArchitectureDiagram.puml]: Resolved

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (Leeyp) [reply docs/DeveloperGuide.md:24]: Resolved

**Review by kwokyto — APPROVED** [review]
LGTM

**Review by SimJJ96 — APPROVED** [review]
LGTM

**Pull request merged**